lints.rs

```diff
@@ -13,36 +13,45 @@
     clippy::exit,
     clippy::explicit_into_iter_loop,
     clippy::filter_map_next,
+    clippy::float_cmp_const,
     clippy::fn_params_excessive_bools,
+    clippy::get_unwrap,
     clippy::if_let_mutex,
     clippy::imprecise_flops,
     clippy::inefficient_to_string,
     clippy::let_unit_value,
     clippy::linkedlist,
     clippy::lossy_float_literal,
     clippy::macro_use_imports,
+    clippy::map_err_ignore,
     clippy::map_flatten,
     clippy::map_unwrap_or,
     clippy::match_on_vec_items,
     clippy::match_wildcard_for_single_variants,
     clippy::mem_forget,
     clippy::mismatched_target_os,
+    clippy::mutex_integer,
     clippy::needless_borrow,
     clippy::needless_continue,
     clippy::option_option,
+    clippy::print_stdout,
     clippy::pub_enum_variant_names,
     clippy::ref_option_ref,
     clippy::rest_pat_in_fully_bound_structs,
+    clippy::string_add,
     clippy::string_to_string,
     clippy::suboptimal_flops,
     clippy::todo,
+    clippy::trivially_copy_pass_by_ref,
     clippy::unnested_or_patterns,
     clippy::unused_self,
+    clippy::useless_let_if_seq,
```

Comment:
like the idea of `useless_let_if_seq` but when tested it before it only found like a single case in our code, so wasn't sure if it was really working correctly. has it been useful for y'all?

Reply:
tested it again, and ran into a bad not compiling suggestion from the lint. so don't think it is mature enough to be enabled generally. https://github.com/rust-lang/rust-clippy/issues/2918#issuecomment-793146825

Reply:
It might have found some mishaps in our code but yeah, given the false suggestions this is not safe to be enabled yet. That looks gross!